lib/operators/operatorDateCompare.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const moment = require('moment');
+
+function getAnswer(questionDotPointer, answers) {
+    let value = false;
+    if (!questionDotPointer) {
+        return value;
+    }
+    const questionId = questionDotPointer.split('.')[3];
+    Object.keys(answers).forEach(page => {
+        if (questionId in answers[page]) {
+            value = answers[page][questionId];
+        }
+    });
+    return value;
+}
+
+function dateCompare(rule, data) {
+    const inputDate = getAnswer(rule[1], data.answers); // LHS.
+    // direction of comparison. e.g. '>' means: is `inputDate` more than `amount` of `units`  different from `compareDate`.
+    // i.e. is '2005-02-01T00:00:00.000Z' more than -12 years different from (i.e. 12 years before) '2021-02-01T00:00:00.000Z' - this evaluates to `true`.
+    const comparator = ['>', '<', '=='].includes(rule[2]) ? rule[2] : '>'; // can be '>', '<', or '=='. defaults to '>'.
+    const amount = parseInt(rule[3], 10); // number of `units` of time that we want to compare the days by e.g. '12', '-3'.
+    const units = rule[4] || 'years'; // can be seconds, minutes, hours, days, weeks, months, or years.
+    // date that you are comparing the `inputDate` against. when
+    // `undefined` is passed in to moment(), it defaults to today's date.
+    // if nothing is defined here, `undefined` is used.
+    const compareDate = getAnswer(rule[5], data.answers) || undefined;
+
+    // get difference, including decimal points (3rd paramater set to `true`).
+    const diff = moment(inputDate).diff(moment(compareDate), units, true);
```

Comment:
Is there a way to reduce the precision of this calculation? We don't record times and default our timestamps to midnight, so having the default level of precision - about 15 decimal places - seems unnecessary and causes most of the other things I've flagged in this review.

Reply:
Added test to demonstrate my thinking here:
https://github.com/CriminalInjuriesCompensationAuthority/q-router/pull/24/commits/3c1871ca4b47bbc61ba0a61de244d9da3c4a0906

Reply:
**ignore my previous comment**

In the interest of not having functionality we don't need (at the moment). I have made it work with integers only